google-oauth-client/src/main/java/com/google/api/client/auth/openidconnect/IdTokenVerifier.java

```diff
@@ -280,5 +402,163 @@ public Builder setAcceptableTimeSkewSeconds(long acceptableTimeSkewSeconds) {
       this.acceptableTimeSkewSeconds = acceptableTimeSkewSeconds;
       return this;
     }
+
+    /** Returns an instance of the {@link Environment} */
+    public final Environment getEnvironment() {
+      return environment;
+    }
+
+    /** Sets the environment. Used mostly for testing */
+    public Builder setEnvironment(Environment environment) {
+      this.environment = environment;
+      return this;
+    }
+
+    /**
+     * Set the HttpTransportFactory used for requesting public keys from the certificate URL. Used
+     * mostly for testing.
+     *
+     * @param httpTransportFactory the HttpTransportFactory used to build certificate URL requests
+     * @return the builder
+     */
+    public Builder setHttpTransportFactory(HttpTransportFactory httpTransportFactory) {
+      this.httpTransportFactory = httpTransportFactory;
+      return this;
+    }
+  }
+
+  /** Custom CacheLoader for mapping certificate urls to the contained public keys. */
+  static class PublicKeyLoader extends CacheLoader<String, Map<String, PublicKey>> {
+    private final HttpTransportFactory httpTransportFactory;
+
+    /**
+     * Data class used for deserializing a JSON Web Key Set (JWKS) from an external HTTP request.
+     */
+    public static class JsonWebKeySet extends GenericJson {
+      @Key public List<JsonWebKey> keys;
+    }
+
+    /** Data class used for deserializing a single JSON Web Key. */
+    public static class JsonWebKey {
+      @Key public String alg;
+
+      @Key public String crv;
+
+      @Key public String kid;
+
+      @Key public String kty;
+
+      @Key public String use;
+
+      @Key public String x;
+
+      @Key public String y;
+
+      @Key public String e;
+
+      @Key public String n;
+    }
+
+    PublicKeyLoader(HttpTransportFactory httpTransportFactory) {
+      super();
+      this.httpTransportFactory = httpTransportFactory;
+    }
+
+    @Override
+    public Map<String, PublicKey> load(String certificateUrl) throws Exception {
+      HttpTransport httpTransport = httpTransportFactory.create();
+      JsonWebKeySet jwks;
+      try {
+        HttpRequest request =
+            httpTransport
+                .createRequestFactory()
+                .buildGetRequest(new GenericUrl(certificateUrl))
+                .setParser(GsonFactory.getDefaultInstance().createJsonObjectParser());
+        HttpResponse response = request.execute();
+        jwks = response.parseAs(JsonWebKeySet.class);
+      } catch (IOException io) {
+        return ImmutableMap.of();
```

Comment:
This doesn't show any error or exceptions when the certificate doesn't have the expected format. Is there.a good way to report the problem?

Reply:
This definitely makes sense, I left it as is since it looks as intentional. I tried to figure out the background for this and it looks like - it is related to LoadingCache implementation, like unable to throw checked exception from runner or something like that. My call is to leave this specific one to avoid as is because: 
1. it does not seem to be a pain point for other library
2. we want to error on the side of reliability at the moment, we can improve experience later
3. library is already officially deprecated

@chingor13 are you ok with that?